updatem/orchestration/update_orchestrator_feedback_test.go

```diff
@@ -0,0 +1,810 @@
+			domains: map[string]types.StatusType{
+				"testDomain1": types.StatusSuperseded,
+			},
+			expectedStatus:       types.StatusRunning,
+			expectedDomainStatus: types.StatusSuperseded,
+			testCode:             func() {},
+		},
+		"test_status_not_from_supported": {
+			handleStatus:     "NOT_SUPPORTED_STATUS",
+			updateOrchStatus: types.StatusRunning,
+			domains: map[string]types.StatusType{
+				"testDomain1": types.StatusSuperseded,
+			},
+			expectedStatus:       types.StatusRunning,
+			expectedDomainStatus: types.StatusSuperseded,
+			testCode:             func() {},
+		},
+	}
+	for testName, testCase := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			t.Log("TestName: ", testName)
+			testUpdOrch := &updateOrchestrator{
+				operation: &updateOperation{
+					activityID:           test.ActivityID,
+					domains:              testCase.domains,
+					desiredStateCallback: eventCallback,
+					actions:              testOperationActions,
+					status:               testCase.updateOrchStatus,
+					identDone:            make(chan bool, 1),
+					identErrChan:         make(chan bool, 1),
+					errChan:              make(chan bool, 1),
+					done:                 make(chan bool, 1),
+					statesPerDomain: map[api.UpdateManager]*types.DesiredState{
+						mockUpdateManager: {},
+					},
+				},
+				cfg: test.CreateTestConfig(false, false),
+			}
+
+			testCase.testCode()
+
+			testUpdOrch.HandleDesiredStateFeedbackEvent("testDomain1", test.ActivityID, "", testCase.handleStatus, "testMsg", testUpdateActions)
+
+			assert.Equal(t, testCase.expectedDomainStatus, testUpdOrch.operation.domains["testDomain1"])
+			assert.Equal(t, testCase.expectedStatus, testUpdOrch.operation.status)
+			assert.Equal(t, testCase.expectedDelayedStatus, testUpdOrch.operation.delayedStatus)
+			assert.Equal(t, testCase.expectedIdentErrMsg, testUpdOrch.operation.identErrMsg)
+		})
+	}
+}
+
+func TestValidateActivity(t *testing.T) {
+	t.Run("test_HandleDesiredStateFeedbackEvent_validateActivity_updateOrchestrator_operation_is_nil", func(t *testing.T) {
+		testUpdOrch := &updateOrchestrator{}
+		testUpdOrch.HandleDesiredStateFeedbackEvent("testDomain", test.ActivityID, "", types.BaselineStatusCleanupFailure, "testMsg", testUpdateActions)
+	})
+	t.Run("test_validateActivity_activityId_missmatch", func(t *testing.T) {
+		testUpdOrch := &updateOrchestrator{
+			operation: &updateOperation{
+				activityID: test.ActivityID,
+			},
+		}
+		assert.False(t, testUpdOrch.validateActivity("testDomain", "difActivityId"))
+	})
+	t.Run("test_validateActivity_domains_missmatch", func(t *testing.T) {
+		testUpdOrch := &updateOrchestrator{
+			operation: &updateOperation{
+				activityID: test.ActivityID,
+				domains: map[string]types.StatusType{
+					"testDomain": types.StatusCompleted,
+				},
+			},
+		}
+		assert.False(t, testUpdOrch.validateActivity("difTestDomain", test.ActivityID))
+	})
+}
+
+func TestDomainUpdateCompleted(t *testing.T) {
```

Comment:
I think this test can also benefit from the table table based approach. The assertions are quite repetative